matter_server/server/device_controller.py

```diff
@@ -969,6 +970,10 @@ def event_callback(
                 data=data.Data,
             )
             self.event_history.append(node_event)
+
+            if isinstance(data.Data, Clusters.BasicInformation.Events.ShutDown):
+                loop.call_soon_threadsafe(self._node_unavailable, node_id)
```

Comment:
can this be grouped in next threadsafe call ?

Reply:
Since we don't return in any of this code, we'll always call back to the main Thread. I've moved all processing to the main thread, this makes it much easier to deal with more changes to event processing in the future.